crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_commitment_helper_gpu.rs

```diff
@@ -0,0 +1,673 @@
+                            [i + scalar_col * single_bit_table_entry.len()];
+                        let end = start + (single_bit_table_entry[i] / BYTE_SIZE) as usize;
+
+                        // For signed offset
+                        let offset_idx = i
+                            + scalar_col * single_packed_byte_with_offset_size
+                            + single_packed_byte_size;
+
+                        let column = &committable_columns[i];
+                        match column {
+                            CommittableColumn::Boolean(column) => {
+                                scalar_row_slice[start..end]
+                                    .copy_from_slice(&column[index].offset_to_bytes());
+                            }
+                            CommittableColumn::TinyInt(column) => {
+                                scalar_row_slice[start..end]
+                                    .copy_from_slice(&column[index].offset_to_bytes());
+
+                                scalar_row_slice[offset_idx] = 1_u8;
+                            }
+                            CommittableColumn::SmallInt(column) => {
+                                scalar_row_slice[start..end]
+                                    .copy_from_slice(&column[index].offset_to_bytes());
+
+                                scalar_row_slice[offset_idx] = 1_u8;
+                            }
+                            CommittableColumn::Int(column) => {
+                                scalar_row_slice[start..end]
+                                    .copy_from_slice(&column[index].offset_to_bytes());
+
+                                scalar_row_slice[offset_idx] = 1_u8;
+                            }
+                            CommittableColumn::BigInt(column)
+                            | CommittableColumn::TimestampTZ(_, _, column) => {
+                                scalar_row_slice[start..end]
+                                    .copy_from_slice(&column[index].offset_to_bytes());
+
+                                scalar_row_slice[offset_idx] = 1_u8;
+                            }
+                            CommittableColumn::Int128(column) => {
+                                scalar_row_slice[start..end]
+                                    .copy_from_slice(&column[index].offset_to_bytes());
+
+                                scalar_row_slice[offset_idx] = 1_u8;
+                            }
+                            CommittableColumn::Scalar(column)
+                            | CommittableColumn::Decimal75(_, _, column)
+                            | CommittableColumn::VarChar(column) => {
+                                scalar_row_slice[start..end]
+                                    .copy_from_slice(&column[index].offset_to_bytes());
+                            }
+                            CommittableColumn::RangeCheckWord(_) => todo!(),
+                        }
```

Comment:
Since this is a large block of very repetitive code, it makes sense to extract this into a separate function to improve readability of the main function.